Chatkit/Data Providers/TypingUsersProvider.swift

```diff
@@ -27,10 +27,13 @@ public class TypingUsersProvider {
         return Set(users)
     }
     
+    let currentUser: User
+    
     // MARK: - Initializers
     
-    init(room: Room, persistenceController: PersistenceController) {
+    init(room: Room, currentUser: User, persistenceController: PersistenceController) {
         self.roomIdentifier = room.identifier
+        self.currentUser = currentUser
```

Comment:
As the other fields of user (name, custom data etc) might change, and won't be updated here, perhaps we should only store the id, as it's all we actually need to identify the user?

Reply:
Also, surely this should be registered in the ViewModel where it is used, not here?

If you're worried about the constructor args for the view model, I think it's either valid to require it, or we can add factory methods for view models to the Chatkit object as Zan suggested

Reply:
One could argue that the same is valid for `typingUsers` property of `TypingUsersProvider`.

Reply:
Aren't the typing users snapshot from the coredata frequently, at least? (but yes, that is a conversation we still need to have!)